### PR TITLE
Add a CognitiveSearch sample

### PR DESCRIPTION
**Pull request by tg-msft**
There are a few hacks and tweaks to the swagger readme, the generator, and the supporting shared source.  It'll be easier to view by commit to catch the meaningful changes.

### CODE REVIEW

**Comment by pakrym** [comment]
Did you manually change the code after generation?

**Comment by tg-msft** [comment]
No, but I was only running `./eng/Generate.ps1 CognitiveSearch`.  Let me try running the entire script and see if that takes care of the diff check.

File: src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs [hunk 1]
```diff
@@ -134,6 +134,12 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                             writer.Append($"({name}")
                                 .AppendNullableValue(implementationType);
 
+                            // TODO: Turn a char into a string so it can be passed to WriteStringValue
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs]: Can you add an overload to https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Azure.Core.Shared/Utf8JsonWriterExtensions.cs#L12

File: src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs [hunk 2]
```diff
@@ -347,6 +353,14 @@ public static void ToDeserializeCall(this CodeWriter writer, JsonValueSerializat
                     }
 
                     writer.AppendRaw(")");
+
+                    // TODO: Turn a GetString() into a char in a safer way
+                    if (frameworkType == typeof(char))
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs]: Can you add a GetChar method to https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Azure.Core.Shared/JsonElementExtensions.cs#L65 and use it higher up

**Comment by tg-msft** [comment]
Filed #419 to track fixing the `request` rename in the swagger README transform.

**Pull request merged**